include/nestegg/nestegg.h

```diff
@@ -427,6 +429,17 @@ int nestegg_packet_discard_padding(nestegg_packet * packet,
     @retval -1 Error.*/
 int nestegg_packet_encryption(nestegg_packet * packet);
 
+/** Query if a packet is partitioned.
+@param packet Packet initialized by #nestegg_read_packet.
+@retval  #NESTEGG_PACKET_HAS_SIGNAL_BYTE_FALSE No signal byte, partition
+information not read from packet.
+@retval  #NESTEGG_PACKET_HAS_SIGNAL_BYTE_UNPARTITIONED Partitioned bit not
+set, partition information not read from packet.
+@retval  #NESTEGG_PACKET_HAS_SIGNAL_BYTE_PARTITIONED Partitioned bit set,
+partition information read from packet.
+@retval -1 Error.*/
+int nestegg_packet_partitioned(nestegg_packet * packet);
```

Comment:
Does it make sense to roll this in to `nestegg_packet_encryption` instead?  It looks like that could be extended to check the partitioned bit and return `PARTITIONED` in that case, and then we could do away with `UNPARTITIONED`.

It does mean that existing users of `nestegg_packet_encryption` need to be changed, but I'm pretty sure Gecko is the only one.